src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultDisplayTemplates.cs

```diff
@@ -224,7 +225,7 @@ public static IHtmlContent ObjectTemplate(IHtmlHelper htmlHelper)
             var viewEngine = serviceProvider.GetRequiredService<ICompositeViewEngine>();
             var viewBufferScope = serviceProvider.GetRequiredService<IViewBufferScope>();
 
-            var content = new HtmlContentBuilder();
+            var content = new HtmlContentBuilder(modelExplorer.Metadata.Properties.Count);
```

Comment:
`Count` isn't perfect because some properties may be ignored in the following loop. But, unless there's exactly 4 properties, this is better than allocating an `object[4]` array and then doubling its size if needed.